kernel/siw_cm.c

```diff
@@ -681,6 +723,39 @@ static int siw_proc_mpareq(struct siw_cep *cep)
 		cep->mpa.hdr.params.c = 1;
 #endif
 
+	cep->p2ptype = SIW_MPAV2_P2P_DISABLED;
+
+	if (__mpa_rr_revision(req->params.bits) == MPA_REVISION_2) {
+		if (req->params.bits & MPA_RR_FLAG_ENHANCED) {
+			cep->enhanced_rdma_connection = 1;
+		} else {
+			cep->enhanced_rdma_connection = 0;
+		}
+	
+		if (cep->enhanced_rdma_connection) {
+			v2 = (struct mpa_v2_data *)cep->mpa.pdata;
+			cep->ird = ntohs(v2->ird) & MPA_IRD_ORD_MASK;
+			cep->ird = min(cep->ird, SIW_MAX_IRD);
+			cep->ord = ntohs(v2->ord) & MPA_IRD_ORD_MASK;
+			cep->ord = min(cep->ord, SIW_MAX_ORD);
+
+			cep->mpa.enhanced_conn_data.ird = htons(cep->ird) | 
```

Comment:
I followed the example of the other MPA enums (e.g. MPA_RR_FLAG_MARKERS) and put them in network byte order. So I think it is or'ing network byte order with network byte order. Please double check me.